packages/core/src/browser/status-bar/status-bar.tsx

```diff
@@ -115,21 +141,26 @@ export class StatusBarImpl extends VirtualWidget implements StatusBar {
                 rightEntries.push(this.renderElement(entry));
             }
         });
-        const leftElements = h.div({ className: 'area left' }, VirtualRenderer.flatten(leftEntries));
-        const rightElements = h.div({ className: 'area right' }, VirtualRenderer.flatten(rightEntries));
-        return VirtualRenderer.flatten([leftElements, rightElements]);
+        const leftElements = <div className="area left"><React.Fragment>{leftEntries}</React.Fragment></div>;
+        const rightElements = <div className="area right"><React.Fragment>{rightEntries}</React.Fragment></div>;
+
+        return [leftElements, rightElements];
+    }
+
+    protected onclick(entry: StatusBarEntry): () => void {
+        return () => {
```

Comment:
It creates a new function each time. We should use exactly the same function for all entries. It can be accomplished by retrieving an entry id from an event target.